pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -168,6 +156,24 @@ func (configgen *ConfigGeneratorImpl) BuildSidecarOutboundHTTPRouteConfig(env mo
 	return out
 }
 
+// buildVirtualHostDomains generates the set of domain matches for a service being accessed from
+// a proxy node
+func buildVirtualHostDomains(service *model.Service, port int, node model.Proxy) []string {
+	domains := []string{service.Hostname, fmt.Sprintf("%s:%d", service.Hostname, port)}
+	if !service.MeshExternal {
```

Comment:
Add a comment like "for external services, don't generate alternate names but use only the user-provided name" or similar? (It's not immediately obvious why this check exists, so lets explain the business logic here for later reference.)

Reply:
Actually I realized that now that we have this logic in place, we no longer need that if block